site/topics/logic/combinational-logic.rst

```diff
@@ -10,6 +10,227 @@ Combinational Logic
 Decoders
 ========
 
+* Consider a situation where an input signal controls two outputs, where only one should be active at a time
+
+    * When the input signal is low, one of the two outputs is high
+    * When the input signal is high, the other output should be high
+
+
+.. list-table:: Single Bit Decoder
+    :widths: auto
+    :align: center
+    :header-rows: 1
+
+    * - Input
+      -
+      - Output a
+      - Output b
+    * - ``0``
+      -
+      - ``1``
+      - ``0``
+    * - ``1``
+      -
+      - ``0``
+      - ``1``
+
+
+.. note::
+
+    Until now, all truth tables were shown to demonstrate how the boolean operators worked, each taking some number of
+    inputs and producing a single output. These truth tables were used to describe how each operator worked.
+
+    Here, the truth table is not describing any specific operator, but some desired functionality. The goal would then
+    be to construct some circuit with logic gates to produce the desired functionality.  Further, it's showing how the
+    desired functionality has more than one output. More specifically, when the input is ``0``, output a should be ``0``
+    and output b should be ``1``, and when the input is ``1``, output a should be ``1`` and output b should be ``0``.
+
+
+* Based on the truth table, how could this functionality be achieved with boolean operators?
+
+    * Output a is the inversion of the input
+    * Output b is simply the input
+
+
+.. figure:: single_bit_decoder.png
+    :width: 500 px
+    :align: center
+
+    A single bit decoder. At any point in time, a single output is always high. The specific output that is high is
+    controlled by the input signal.
+
+
+* This is an example of a single bit decoder
+
+    * The one input signal can be thought of as a single bit
+    * The input signal is "decoded" to control the output signals
+
+
+More than One Input Bit
+-----------------------
+
+* Now consider a situation where four separate circuits need to be controlled by a decoder
+
+    * These circuits will be labeled 0 to 3 for ease
```

Comment:
This is enough motivation imo. They've written some code by now, so the utility of conditional branching should be clear to them.  
The other approach I can think of is to do memory -> decoder -> multiplexing (or swap memory and multiplexing). So like, a concrete problem -> a general tool to solve it -> another application of that tool. Though, I feel multiplexors have the same kind of /shrug as decoders, so either would still feel like decoders=memory to some extend.

Reply:
I may go this way. I'll get the next few topics down and then look to see if they can be massaged into place better